analyzer/src/main/kotlin/managers/Yarn2.kt

```diff
@@ -0,0 +1,141 @@
+        override val globsForDefinitionFiles = listOf("package.json")
+
+        override fun create(
+            analysisRoot: File,
+            analyzerConfig: AnalyzerConfiguration,
+            repoConfig: RepositoryConfiguration
+        ) = Yarn2(managerName, analysisRoot, analyzerConfig, repoConfig)
+    }
+
+    /**
+     * The Yarn2+ is not installed globally: we use the program shipped by the project in .yarn/releases. The resource
+     * file `.yarnrc.yml` contains the path to the executable under the property name `yarnPath` defined
+     * for the current project e.g. yarnPath: .yarn/releases/yarn-3.2.1.cjs.
+     * This map holds the mapping between the directory and their Yarn2 executables.
+     */
+    private val yarn2CommandsByPath: MutableMap<File, String> = mutableMapOf()
+
+    /**
+     * A NPM package manager, used to call NPM when Yarn2+ does not provide enough information.
+     */
+    private val delegateNpmAnalyzer = Npm("npm", analysisRoot, analyzerConfig, repoConfig)
+
+    companion object {
+        const val YARN2_RESOURCE_FILE = ".yarnrc.yml"
+
+        // The property in `.yarnrc.yml`containing the path to the Yarn2+ executable.
+        const val YARN_PATH_PROPERTY_NAME = "yarnPath"
+    }
+
+    override fun hasLockFile(projectDir: File) = hasYarnLockFile(projectDir)
+
+    override fun command(workingDir: File?): String {
+        return workingDir?.let {
+            yarn2CommandsByPath.getOrPut(it) {
+                val yarnConfig = yamlMapper.readTree(workingDir.resolve(YARN2_RESOURCE_FILE))
+                val yarnCommand = yarnConfig.get(YARN_PATH_PROPERTY_NAME)
+                yarnCommand?.let { command ->
+                    val yarnExecutable = workingDir.resolve(command.textValue())
+
+                    // TODO: Yarn2 executable is a `cjs` file. Check if under Windows it needs to be run with `node`.
+
+                    // TODO: This is a security risk to blindly run code coming from a repository other than ORT's. ORT
+                    //  should download the Yarn2 binary from the official repository and run it.
+                    require(yarnExecutable.exists()) {
+                        "The Yarn2+ program '${yarnExecutable.name}' does not exist."
+                    }
+
+                    if (!yarnExecutable.canExecute()) {
+                        log.warn {
+                            "The Yarn2+ program '${yarnExecutable.name}' should be executable. Changing its rights."
+                        }
+                        require(yarnExecutable.setExecutable(true)) {
+                            "Cannot set the Yarn2+ program to be executable."
+                        }
+                    }
+                    yarnExecutable.absolutePath
+                } ?: throw IllegalStateException("No Yarn2+ executable could be found in 'yarnrc.yml'.")
+            }
+        } ?: "yarn2+ executable' in .yarn/releases/ and configured in .yarnrc.yml"
```

Comment:
This is an awkward string to return here, as it's not a command to execute.

Reply:
This is because `workingDir` is `null` only when called by `ort requirements`.

Therefore this output is made for the output of `ort requirements`

```
	- Stack: Requires 'stack' in version >=2.1.1. Tool not found.
	- Yarn: Requires 'yarn' in version >=1.3.0 <1.23.0. Tool not found.
	- Yarn2: Requires 'yarn2+ executable' in .yarn/releases/ and configured in .yarnrc.yml' in version >=2.0.0. Tool not found.
```

Since this is an ORT quirk, I am open for opinion here.

Reply:
IMO, Yarn2 should not be mentioned at all in the `requirements` command, as no installation is required.
`Gradle` is also not mentioned in `requirements`. Then this String could also be removed.

I think it's dangerous to return this String as a command. I think you cannot rely on `command` without a `workingDir` will only be called from `requirements`.